libmamba/include/mamba/api/install.hpp

```diff
@@ -35,11 +34,26 @@ namespace mamba
         const Configuration& config,
         const std::vector<std::string>& specs,
         bool create_env = false,
-        bool remove_prefix_on_failure = false,
-        int solver_flag = SOLVER_INSTALL,
-        int is_retry = 0
+        bool remove_prefix_on_failure = false
+    );
+
+    auto create_install_request(  //
+        PrefixData& prefix_data,
+        std::vector<std::string> specs,
+        bool freeze_installed
+    ) -> solver::Request;
+
+    void add_pins_to_request(
+        solver::Request& request,
+        const Context& ctx,
+        PrefixData& prefix_data,
+        std::vector<std::string> specs,
+        bool no_pin,
+        bool no_py_pin
```

Comment:
- What does `no_py_pin` means, and why do we have a specific handling for it?
- Can we imagine that in the future, more specific pinning handling would be required? In that case, maybe passing a "pin" structure instead of a bunch of boolean values coulb be safer.
- no_pin => no_gin =====>[] :smile:

Reply:
It's one of the CLI option ` --no-py-pin,--py-pin{false} Do not automatically pin Python`. Why? Conda.
Yes, we should definitly handle all that Python stuff in a more generic way, also true for `pip_as_python_dependency` things which should really be a `recommend` or something.

For now I'm just trying to change the solver.